### PR TITLE
Add ADNI dataset Siamese network/classifier model

### PR DESCRIPTION
**Pull request by drobson03**
This PR adds a Siamese Network/Classifier model trained on the ADNI dataset.

On the (entire) test dataset, this model, unfortunately, does not meet the minimum accuracy requirement of 80% and only achieves 67.12%, though this could be improved with further hyperparameter/architecture tuning.

The following files are included in this PR under the `recognition/46991030-ADNI-siamese` directory:

- `README.md` - an overview of the problem, model, dependencies and how to use, etc.
- `constants.py` - constants including image shape, dataset path and model output paths
- `dataset.py` - functions to load the dataset from the given path
- `modules.py` - components of the architecture for the model(s).
- `predict.py` - a script to sample some test data and display results visually.
- `train.py` - a script to load the data, train/validate each model and test the models on the datasets.
- `utils.py` - contains a function to plot the training history of models

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Good
- clear flow of information
- algorithm and usage explained well
- relevant plots/figures provided
- results followed with discussion

Commit messages: Good

Code: 
- good design
- minimal to no comments
- algorithm is implemented correctly

Functionality/Performance: 
- both siamese and classifier seem to be massively overfitting, although no mention of steps taken to try and resolve this
- classifier only achieves 67% accuracy on the test set

General comments:
- algorithm solves the problem apporpriately although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good, refs missing -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
